<?xml version="1.0" encoding="utf-8"?>
<robot name="valve" xmlns:xacro="http://www.ros.org/wiki/xacro">
    <xacro:arg name="resource_directory" default="$(find mppi_omav_interaction)/data/meshes"/>
    <xacro:arg name="object_position" default="-1 0 1"/>
    <xacro:arg name="object_rpy" default="1.5708 0 0"/>
    <xacro:arg name="valve_type" default="lee"/>

    <xacro:property name="base_position_x" default="1.0"/>
    <xacro:property name="base_position_y" default="0.0"/>
    <xacro:property name="base_length" default="0.2"/>
    <xacro:property name="base_height" default="0.5"/>
    <xacro:property name="base_width" default="0.2"/>

    <link name="world"/>

    <!-- <joint name="fixed" type="fixed">
        <parent link="world"/>
        <child link="valve_base"/>
        <origin xyz="${base_position_x} ${base_position_y} ${base_height/2.0}" rpy="0.0 0 0"/>
    </joint>

    <link name="valve_base">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="${base_width} ${base_length} ${base_height}"/>
            </geometry>
            <material name="base_material">
                <color rgba="1.0 1.0 1.0 1"/>
            </material>
        </visual>
    </link> -->

    <link name="handle_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="file://$(arg resource_directory)/valve_$(arg valve_type)_visual.obj" scale="1 1 1"/>
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
                <mesh  filename="file://$(arg resource_directory)/valve_$(arg valve_type)_collision.obj" scale="1 1 1"/>
            </geometry>
        </collision>
    </link>

    <joint name="articulation_joint" type="revolute">
        <parent link="world"/>
        <child link="handle_link"/>
        <origin xyz="$(arg object_position)" rpy="$(arg object_rpy)"/>
        <dynamics damping="0.0" friction="0.0"/>
        <axis xyz="0 0 1"/>
        <limit upper="300.14" lower="-300.14" effort="100" velocity="1.0"/>
    </joint>

</robot>
